components/authz-service/storage/v2/memstore/memstore.go

```diff
@@ -312,24 +312,39 @@ func (s *State) ListRules(_ context.Context) ([]*storage.Rule, error) {
 	return rules, nil
 }
 
-func (s *State) ListRulesForProject(_ context.Context, projectID string) ([]*storage.Rule, error) {
+func (s *State) ListRulesForProject(_ context.Context, projectID string) ([]*storage.Rule, storage.ProjectRulesStatus, error) {
 	_, exists := s.projects.Get(projectID)
 	if !exists {
-		return nil, storage_errors.ErrNotFound
+		return nil, storage.RulesStatusError, storage_errors.ErrNotFound
 	}
 
 	items := s.rules.Items()
 	rules := []*storage.Rule{}
 
+	anyStagedRules := false
 	for _, item := range items {
 		if rule, ok := item.Object.(*storage.Rule); ok {
 			if rule.ProjectID == projectID {
+				if rule.Status == "staged" {
+					anyStagedRules = true
+				}
 				rules = append(rules, rule)
 			}
 		}
 	}
 
-	return rules, nil
+	var rulesStatus v2.ProjectRulesStatus
+	if len(rules) == 0 {
+		rulesStatus = v2.NoRules
+	} else {
+		if anyStagedRules {
+			rulesStatus = v2.EditsPending
+		} else {
+			rulesStatus = v2.Applied
```

Comment:
I suppose you could skip this branch of the conditional and set the var to `v2.Applied` at the start (on 336)

Reply:
That seems less clear to me on what the logic is personally. Curious what others think.

Reply:
Go convention seems to be drop the `else` where possible

Reply:
Ok I can do it that way, I think it's a bit harder to parse though